templates/documents/shared/details_base.html

```diff
@@ -0,0 +1,28 @@
+<form
+    id="detail-form"
+    action="{{ edit_url }}"
+    method="POST"
+    class="m-2 grid grid-cols-4" x-data="{editing: false}">
+    {% csrf_token %}
+    {% block detail_content %}
+    {% endblock %}
+    <div class="py-3 col-span-1">
+        <div class="join">
+            <button type="button" class="btn btn-sm join-item pl-3" x-show="!editing" @click="editing = true">
+                Edit <i class="fa-solid fa-pencil fa-xs"></i>
+            </button>
+            <button type="submit" class="btn btn-sm join-item" x-cloak x-show="editing">
+                Update <i class="fa-solid fa-check fa-xs"></i>
+            </button>
+            <a type="button" class="btn btn-sm join-item" x-show="!editing" href="{{ delete_url }}">
+                <i class="fa-solid fa-trash htmx-hide"></i>
+            </a>
+            <button type="button" @click="editing = false" class="btn btn-sm join-item" x-cloak x-show="editing">
+                Cancel
```

Comment:
Add icon for consistency
```suggestion
                Cancel <i class="fa-solid fa-ban fa-xs"></i>
```

I think all buttons should be the same so all have text & icon, or only text, or only icon. My preference is only icons for a button bar like this.